tests/unit/maze_dataset/tokenization/test_token_utils.py

```diff
@@ -498,3 +520,471 @@ class Z:
+                        (DC8(False), True),
+                        (DC8(True), True),
+                        (DC9(False, False), True),
+                        (DC9(False, True), True),
+                        (DC9(True, False), True),
+                        (DC9(True, True), True),
+                        (DC8(False), False),
+                        (DC8(True), False),
+                        (DC9(False, False), False),
+                        (DC9(False, True), False),
+                        (DC9(True, False), False),
+                        (DC9(True, True), False),
+                    ],
+                ),
+            ]
+        )
+    ],
+)
+def test_all_instances(
+    type_: FiniteValued, result: type[Exception] | Iterable[FiniteValued]
+):
+    if isinstance(result, type) and issubclass(result, Exception):
+        with pytest.raises(result):
+            all_instances(type_)
+    elif hasattr(type_, "__dataclass_fields__"):
+        assert dataclass_set_equals(all_instances(type_), result)
+    else:
+        assert set(all_instances(type_)) == set(result)
+
+
+# @mivanit: this was really difficult to understand
+@mark.parametrize(
+    "type_, validation_funcs, assertion",
+    [
+        param(
+            type_,
+            vfs,
+            assertion,
+            id=f"{i}-{type_.__name__}",
+        )
+        for i, (type_, vfs, assertion) in enumerate(
+            [
+                (
+                    # type
+                    PathTokenizers._PathTokenizer,
+                    # validation_funcs
+                    frozendict.frozendict({}),
+                    # assertion
+                    lambda x: PathTokenizers.StepSequence(
+                        step_tokenizers=(StepTokenizers.Distance(),)
+                    )
+                    in x,
+                ),
+                (
+                    # type
+                    PathTokenizers._PathTokenizer,
+                    # validation_funcs
+                    frozendict.frozendict(
+                        {
+                            PathTokenizers._PathTokenizer: lambda x: x.is_valid(),
+                        }
+                    ),
+                    # assertion
+                    lambda x: PathTokenizers.StepSequence(
+                        step_tokenizers=(StepTokenizers.Distance(),)
+                    )
+                    not in x
+                    and PathTokenizers.StepSequence(
+                        step_tokenizers=(
+                            StepTokenizers.Coord(),
+                            StepTokenizers.Coord(),
+                        )
+                    )
+                    not in x,
+                ),
+            ]
+        )
+    ],
+)
+def test_all_instances2(
+    type_: FiniteValued,
+    validation_funcs: frozendict.frozendict[
+        FiniteValued, Callable[[FiniteValued], bool]
+    ],
+    assertion: Callable[[list[FiniteValued]], bool],
+):
+    # TODO: error here
```

Comment:
This error is due to `StepTokenizers.Coord() == StepTokenizers.Distance()` returning `True`. I believe the test was passing at commit 26db9f6af701fd492a565854082d2fec6e23c71a, so it should have been returning `False` at that point. It looks like `muutils 0.5.12` (specifically commit 9ee8b08) reversed this behavior.